cql3/functions/aggregate_fcts.cc

```diff
@@ -75,15 +76,66 @@ class count_rows_function final : public native_aggregate_function {
     }
 };
 
+// We need a wider accumulator for sum and average,
```

Comment:
The example of 2147483647 + 1 -1 is interesting. With this implementation we promise that we support intermediate values overflowing, as long as the final value is in range.

If we don't want to support that we could use  __builtin_add_overflow. I am OK with both, just checking.

Reply:
We want to support intermediate overflow to minimize user disruption.

Reply:
Certainly it's important for average. avg(tinyint) always fits in a tinyint, but the intermediate accumulator can easily overflow.

For sum() it's not important, but we have it for free so why not.